src/awx_plugins/credentials/aws_assumerole.py

```diff
@@ -0,0 +1,128 @@
+    """Caught by AnsibleAWSModule."""
+
+_aws_cred_cache = {}
+
+
+assume_role_inputs = {'fields': [{'id': 'access_key',
+                                  'label': _('AWS Access Key'),
+                                  'type': 'string',
+                                  'secret': True,
+                                  'help_text': _('The optional AWS access key for the user who will assume the role'),
+                                  },
+                                 {'id': 'secret_key',
+                                  'label': 'AWS Secret Key',
+                                  'type': 'string',
+                                  'secret': True,
+                                  'help_text': _('The optional AWS secret key for the user who will assume the role'),
+                                  },
+                                 {'id': 'external_id',
+                                  'label': 'External ID',
+                                  'type': 'string',
+                                  'help_text': _('The optional External ID which will be provided to the assume role API'),
+                                  },
+                                 {'id': 'role_arn',
+                                  'label': 'AWS ARN Role Name',
+                                  'type': 'string',
+                                  'secret': True,
+                                  'help_text': _('The ARN Role Name to be assumed in AWS'),
+                                  },
+                                 ],
+                      'metadata': [{'id': 'identifier',
+                                    'label': 'Identifier',
+                                    'type': 'string',
+                                    'help_text': _('The name of the key in the assumed AWS role to fetch [AccessKeyId | SecretAccessKey | SessionToken].'),
+                                    },
+                                   ],
+                      'required': ['role_arn'],
+                      }
+
+
+def aws_assumerole_getcreds(access_key, secret_key, role_arn, external_id):
+    if (access_key is None or len(access_key) == 0) and (
+            secret_key is None or len(secret_key) == 0
+    ):
+        # Connect using credentials in the EE
+        connection = boto3.client(service_name='sts')
+    else:
+        # Connect to AWS using provided credentials
+        connection = boto3.client(
+            service_name='sts',
+            aws_access_key_id=access_key,
+            aws_secret_access_key=secret_key,
+        )
+    try:
+        response = connection.assume_role(
+            RoleArn=role_arn,
+            RoleSessionName='AAP_AWS_Role_Session1',
+            ExternalId=external_id,
+        )
+    except ClientError as ce:
+        raise ValueError(f'Got a bad client response from AWS: {ce.msg}.')
+
+    credentials = response.get('Credentials', {})
+
+    return credentials
+
+
+def aws_assumerole_backend(**kwargs):
```

Comment:
Plz type this too: https://typing.rtfd.io/en/latest/spec/callables.html#unpack-kwargs / https://docs.python.org/3.11/library/typing.html#typing.Unpack.
*(can it have regular explicit args listed instead?)*